src/image/loading_displaying.js

```diff
@@ -387,13 +387,13 @@ p5.prototype.image = function(
 
   p5._validateParameters('image', arguments);
 
-  let defW = img.width;
-  let defH = img.height;
+  let defW = img.width || img.elt.width;
+  let defH = img.height || img.elt.height;
 
   if (img.elt && img.elt.videoWidth && !img.canvas) {
     // video no canvas
-    defW = img.elt.videoWidth;
-    defH = img.elt.videoHeight;
+    defH = defH || img.elt.videoHeight;
+    defW = defW || img.elt.videoWidth;
   }
```

Comment:
any reason H and W are flipped? also, I personally prefer

```javascript
defW = defW !== undefined ? defW : img.elt.videoWidth;
```

just for the weird edge case when dimensions are 0 (same for the lines above)

Reply:
I mistakenly flipped those, now I have fixed it and also made the changes to checking for `undefined` values. The edge case of dimensions having 0 value can definitely come up. Thanks for pointing out this case!